[tp] fix PrepareModuleInput for multiple inputs

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112204

Not all inputs needs to annotate shardings and convert to DTensors, if
user annotate only one inputs are mark the rest as Nones, we should skip
creating DTensors

wanchaol committed:
[tp] fix PrepareModuleInput for multiple inputs

Not all inputs needs to annotate shardings and convert to DTensors, if
user annotate only one inputs are mark the rest as Nones, we should skip
creating DTensors

[ghstack-poisoned]

fduwjj reviewed (approved):
Thanks for fixing it.

wanchaol commented:
@pytorchbot merge

wanchaol committed:
Update on "[tp] fix PrepareModuleInput for multiple inputs"


Not all inputs needs to annotate shardings and convert to DTensors, if
user annotate only one inputs are mark the rest as Nones, we should skip
creating DTensors

[ghstack-poisoned]

wanchaol committed:
Update on "[tp] fix PrepareModuleInput for multiple inputs"


Not all inputs needs to annotate shardings and convert to DTensors, if
user annotate only one inputs are mark the rest as Nones, we should skip
creating DTensors

[ghstack-poisoned]

wanchaol commented:
@pytorchbot merge
